ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditTextClickDetector.java

```diff
@@ -0,0 +1,73 @@
+package com.facebook.react.views.textinput;
+
+import android.view.MotionEvent;
+import android.view.View;
+
+import androidx.annotation.Nullable;
+
+class ReactEditTextClickDetector {
```

Comment:
i elected to make this (and name it) very specific to this edittext. trying to generify it and expose a hook for client apps to inject their own behaviors here is def noble but i don't want to worry about unintended consequences of future changes.